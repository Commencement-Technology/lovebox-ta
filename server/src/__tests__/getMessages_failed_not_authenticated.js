const { gql } = require('apollo-server');
const { createTestClient } = require('apollo-server-testing');
const { ApolloServer, schema, redisClient } = require('../server');
import consola from 'consola';

const server = new ApolloServer({ ...schema, uploads: false });
redisClient
    .connect()
    .then(() =>
        consola.ready({
            message: 'Redis client is connected successfully',
            badge: true,
        })
    )
    .catch(error =>
        consola.error({
            message: `An error occured when connecting Redis client ${error}`,
            badge: true,
        })
    );
const GET_MESSAGES = gql`
    query {
        messages {
            photoUrl
            id
        }
    }
`;

describe('Get messages without token', () => {
    it('Should return error not allowed to access the resource', async () => {
        const { query } = createTestClient(server);
        const res = await query({ query: GET_MESSAGES });
        expect(res.data.messages).toBeNull();
        expect(res.errors).toBeDefined();
        expect(res.errors[0].message).toBe(
            'Your not allowed to access this resource'
        );
        expect(res).toMatchSnapshot();
    });
});
