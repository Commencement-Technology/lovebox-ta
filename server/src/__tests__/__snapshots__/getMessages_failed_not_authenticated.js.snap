// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get messages without token Should return error not allowed to access the resource 1`] = `
Object {
  "data": Object {
    "messages": null,
  },
  "errors": Array [
    [GraphQLError: Your not allowed to access this resource],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
